src/main/java/io/kestra/plugin/serdes/avro/AvroDeserializer.java

```diff
@@ -113,18 +113,27 @@ private static Object objectDeserializer(Object value, Schema schema) {
     }
 
     private static Object unionDeserializer(Object value, Schema schema) {
-        return schema
-            .getTypes()
-            .stream()
-            .map(type -> {
-                try {
-                    return AvroDeserializer.objectDeserializer(value, type);
-                } catch (Exception e) {
-                    return  false;
-                }
-            })
-            .findFirst()
-            .orElseThrow();
+        // first, if value is null, check if the null type exist to avoid generating a NPE
+        if(value == null) {
+            if(schema.getTypes().stream().map(s -> s.getType()).anyMatch(t -> Type.NULL == (t))) {
```

Comment:
```suggestion
            if(schema.getTypes().stream().filter(s -> s.getType() == Type.NULL == (t)).count() > 0) {
```
seems to be more readable no? I need to read the line twice

Reply:
```suggestion
            if (schema.getTypes().stream().anyMatch(t -> t.getType() == Type.NULL)) {
```
ok this one is more readable IMO

Reply:
Agreed